packages/story-editor/src/components/form/tags/input.js

```diff
@@ -135,48 +171,110 @@ function Input({
           setIsInputFocused(false);
         },
       }),
-      [onUndo]
+      [onUndo, suggestedTerms]
     );
 
   const renderedTags = tagBuffer || tags;
+
+  const handleTagSelectedFromSuggestions = useCallback((e, selectedValue) => {
+    e.preventDefault();
+    dispatch({ type: ACTIONS.SUBMIT_VALUE, payload: selectedValue });
+    inputRef?.current.focus();
+  }, []);
+
+  const totalSuggestions = suggestedTerms.length - 1;
+
+  const handleSuggestionKeyDown = useCallback(
+    (e, index, name) => {
+      const nextChild = index + 1;
+      const previousChild = index - 1;
+      if (e.key === 'ArrowDown') {
+        if (nextChild <= totalSuggestions) {
+          menuRef?.current?.children?.[nextChild]?.focus();
+        }
+      }
+      if (e.key === 'ArrowUp') {
+        if (previousChild < 0) {
+          inputRef?.current.focus();
+        } else {
+          menuRef?.current?.children?.[previousChild]?.focus();
+        }
+      }
+      if (e.key === 'Enter') {
+        handleTagSelectedFromSuggestions(e, name);
+      }
+    },
+    [handleTagSelectedFromSuggestions, totalSuggestions]
+  );
+
   return (
-    <Border isInputFocused={isInputFocused}>
-      {
-        // Text input should move, relative to end, with offset
-        // this helps with natural tab order and visuals
-        // as you ArrowLeft or ArrowRight through tags
-        [
-          ...renderedTags.slice(0, renderedTags.length - offset),
-          INPUT_KEY,
-          ...renderedTags.slice(renderedTags.length - offset),
-        ].map((tag) =>
-          tag === INPUT_KEY ? (
-            <TextInput
-              {...props}
-              key={INPUT_KEY}
-              value={value}
-              onKeyDown={handleKeyDown}
-              onChange={handleChange}
-              onFocus={handleFocus}
-              onBlur={handleBlur}
-              size="4"
-            />
-          ) : (
-            <Tag key={tag} onDismiss={removeTag(tag)}>
-              {tagDisplayTransformer(tag) || tag}
-            </Tag>
+    <Border ref={containerRef} isInputFocused={isInputFocused}>
+      <InputWrapper>
+        {
+          // Text input should move, relative to end, with offset
+          // this helps with natural tab order and visuals
+          // as you ArrowLeft or ArrowRight through tags
+          [
+            ...renderedTags.slice(0, renderedTags.length - offset),
+            INPUT_KEY,
+            ...renderedTags.slice(renderedTags.length - offset),
+          ].map((tag) =>
+            tag === INPUT_KEY ? (
+              <TextInput
+                {...props}
+                key={INPUT_KEY}
+                value={value}
+                onKeyDown={handleKeyDown}
+                onChange={handleChange}
+                onFocus={handleFocus}
+                onBlur={handleBlur}
+                size="4"
+                ref={inputRef}
+                autoComplete="off"
+              />
```

Comment:
💭 I think we should change the [`autocomplete`](https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/autocomplete) property to "on" when the terms are being auto-completed. This should help with accessibility

💭 We should also add an `aria-expanded` and an `aria-describedby` to associate this to the autocomplete 

```suggestion
              <TextInput
                {...props}
                key={INPUT_KEY}
                value={value}
                onKeyDown={handleKeyDown}
                onChange={handleChange}
                onFocus={handleFocus}
                onBlur={handleBlur}
                size="4"
                ref={inputRef}
                autoComplete={value.length > 3 ? "on" : "off"}
                aria-describedby={/* id of suggestion list */}
              />
```

Reply:
nice catch, updated it to sync with our new bff the gutenberg taxonomy component 
![Screen Shot 2021-10-01 at 1 51 28 PM](https://user-images.githubusercontent.com/10720454/135685064-77915b11-335b-45db-9434-dfa14356393a.png)